src/System.Management.Automation/engine/remoting/commands/PSRemotingCmdlet.cs

```diff
@@ -445,6 +445,40 @@ internal enum VMState
             FastSavingCritical,
         }
 
+#nullable enable
+        /// <summary>
+        /// Get the State property from Get-VM result.
+        /// </summary>
+        /// <param name="value">The raw PSObject as returned by Get-VM.</param>
+        /// <returns>The VMState value of the State property if present and parsable, otherwise null.</returns>
+        internal VMState? GetVMStateProperty(PSObject value)
+        {
+            ReadOnlyPSMemberInfoCollection<PSPropertyInfo> stateProps = value.Properties.Match("State");
+            if (stateProps.Count == 0)
+            {
+                return null;
+            }
+
+            object? rawState = stateProps[0].Value;
+            if (rawState is Enum enumState)
+            {
+                // If the Hyper-V module was directly importable we have the VMState enum
+                // value which we can just cast to our VMState type.
+                return (VMState)enumState;
+            }
```

Comment:
Will this work?
```suggestion
            if (rawState is VMState result)
            {
                // If the Hyper-V module was directly importable we have the VMState enum
                // value which we can just cast to our VMState type.
                return result;
            }
```

Reply:
Unfortunately no, `VMState` is an enum defined in this class whereas the `Hyper-V` module returns its own enum type. So this is checking if the value is an enum and casts it to the copy of it here. The alternative is making a dep on whatever library the Hyper-V modules are using which I don't think is really possible.

Reply:
That's good to know.